pagefaultgames/pokerogue
# [Dependencies] Update to resolve install warnings

## What are the changes?

As requested by @Tempo-anon I've cleaned the dependencies up. 
Meaning version bumps, remove unused etc.

closes #2353 


> ### ℹ️  After merging this
> we should consider adding dependabot to the project
> see #2433

## Why am I doing these changes?

Mainly to close vulnerabilities and to resolve warnings thrown during the `npm i` run.

## What did change?



1.  ⬆️  [bump up vitest and eslint version to resolve inflight warning](https://github.com/pagefaultgames/pokerogue/commit/5f6712888e23f4e6a33e6778797c56397db86d2e)
    - ❌  remove and disable `eslint-plugin-import` due to not supporting `eslint@9` yet (can be reactivated later) 
    - ⬆️ bump `eslint` to `9.7.0`.
      - ⬆️ bump `@typescript-eslint/eslint-plugin` to `8.0.0-alpha.54`
      - ⬆️ update `@typescript-eslint/parser` to `8.0.0-alpha.54`
      - ⬆️ update `typescript-eslint` to `8.0.0-alpha.54`
      - ❌ remove `eslint-plugin-import` as it's not compatible with eslint 9 yet => see https://github.com/import-js/eslint-plugin-import/issues/2948
    - ⬆️ update `vitest` to `2.0.4`
      - ⬆️ update `@vitest/coverage-istanbul` to `2.0.4`
2.  ❌ [remove vite-plugin-fs dependency...](https://github.com/pagefaultgames/pokerogue/commit/22c68224ecd16b8b58b9b536fd86b417fb340cf1)
    - removed `api-generatort-script.ts` file (this was in consultations with devs [discord-message]( https://discord.com/channels/1125469663833370665/1250836282926436413/1266469312609259550))
3. ⬆️ [bump vite to version 5](https://github.com/pagefaultgames/pokerogue/commit/3e22cdd8cf4c9c5e4d59aff2ad4b3b39481605f7)
4.  🔀  [migrate some typescript-eslint to @stylistic/eslint-plugin-ts](https://github.com/pagefaultgames/pokerogue/pull/3160/commits/71403cecaeb0b39ac94f71cda5e3a5373b926e56)
     - _as stated here: https://typescript-eslint.io/blog/deprecating-formatting-rules#upgrading-to-eslint-stylistic_
5.  ❌ [delete pokenode-ts dependency](https://github.com/pagefaultgames/pokerogue/pull/3160/commits/c5e587e11b7db1908656ae6aef4361ffe60e8925) (unused)
6.  ❌ [remove axios and axios-cache-interceptor dependency](https://github.com/pagefaultgames/pokerogue/pull/3160/commits/49d6ddcec1c78e95e8f4caa3f59e2ef05c7dcdb2) (unused, DOM integrated `fetch` is being used)
7.  ❌ [remove unused json-beautify dependency](https://github.com/pagefaultgames/pokerogue/pull/3160/commits/359e13d09d4d684d24b2a3ec57e4263f0c68db37)
8. 🔁  [move dependency-cruiser to dev dependencies](https://github.com/pagefaultgames/pokerogue/pull/3160/commits/c20c7d59d9f0c9790d50c87280c67657e8417e0e)

### Screenshots/Videos

<img width="386" alt="image" src="https://github.com/user-attachments/assets/62d6556c-2e3b-437b-8dc4-ae65398cad72">
<img width="478" alt="image" src="https://github.com/user-attachments/assets/6bb7f8e2-7fae-41bb-baf8-225c6ce2cf81">
<img width="345" alt="image" src="https://github.com/user-attachments/assets/b576a2a3-ffd7-43c5-9123-f33848005553">

## How to test the changes?

- run `npm i` or `npm ci` => no warnings. No vulnerabilities.
- run `npm run build` => should still work as expected
- run `npm run test:silent` => shouldn't fail
- run `npm run docs` => no errors (plenty of warnings. That's normal)
- run `npm run eslint-ci` => no warnings
- run `npm run start` => should start as expected

## Checklist
- ~~[] Have I considered writing automated tests for the issue?~~
- ~~[] Are the changes visual?~~
  - ~~[] Have I provided screenshots/videos of the changes?~~
